app/controllers/api/v0/signatures_controller.rb

```diff
@@ -1,21 +1,38 @@
 class Api::V0::SignaturesController < ApiController
   def create
     manifest = find_manifest(params[:manifest_id])
-    sign_request = read_body_as_json
+    signature_request = prep_signature_request(manifest)
 
-    manifest_content = manifest[:content].to_json
-    sign_request[:manifest_content] = manifest_content
-    system_session_token = session[:system_session_token]
-    sign_request["token"] = system_session_token
+    cdx_start = Time.now
+    cdx_response = CDX::Manifest.new(signature_request).sign
+    cdx_stop = Time.now
+    Rails.logger.debug(ANSI.blue{ "  CDX signature time: #{sprintf('%#g', (cdx_stop - cdx_start))} seconds" })
 
-    response = CDX::Manifest.new(sign_request).sign
+    status = update_manifest(cdx_response, signature_request, manifest)
 
-    if (response.key?(:document_id))
-      manifest[:document_id] = response[:document_id]
-      manifest[:activity_id] = sign_request["activity_id"]
-      manifest.save!
+    render(json: cdx_response.to_json, status: status) unless performed?
```

Comment:
I usually prefer `if` statements to be indented rather than at the end of a line like this. more code, but easier to read imo.